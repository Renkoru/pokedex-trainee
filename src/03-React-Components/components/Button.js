import React from 'react';
import classNames from 'classnames';

const sizeMap = {
  default: '',
  small: 'is-small',
  normal: 'is-normal',
  medium: 'is-medium',
  large: 'is-large',
};

const colorMap = {
  default: 'is-success',
  primary: 'is-primary',
  link: 'is-link',
  info: 'is-info',
  success: 'is-success',
  warning: 'is-warning',
  danger: 'is-danger',
};

function Button({ children, as, type, size, ...rest }) {
  const DomContainer = as;

  return (
    <DomContainer className={classNames('button', sizeMap[size], colorMap[type])} {...rest}>
      {children}
    </DomContainer>
  );
}

Button.defaultProps = {
  type: 'default',
  size: 'default',
  as: 'button',
};

export default Button;
