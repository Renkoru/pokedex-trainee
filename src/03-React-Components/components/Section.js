import React from 'react';
import PropTypes from 'prop-types';

const Section = ({ children }) => <div className="section">{children}</div>;

Section.propTypes = {
  children: PropTypes.node,
};

export default Section;
