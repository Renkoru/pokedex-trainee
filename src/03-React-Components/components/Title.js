import React from 'react';
import classNames from 'classnames';

function Title({ children, type }) {
  const isSubtitle = type === 'subtitle';
  return (
    <div
      className={classNames({
        title: !isSubtitle,
        subtitle: isSubtitle,
      })}
    >
      {children}
    </div>
  );
}

Title.defaultProps = {
  type: 'title',
};

export default Title;
