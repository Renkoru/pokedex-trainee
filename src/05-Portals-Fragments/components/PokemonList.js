import isEmpty from 'lodash/fp/isEmpty';

import React from 'react';
import styled from '@emotion/styled';
import capitalize from 'lodash/fp/capitalize';

function PokemonList({ title, pokemonList, className, onPokemonClick }) {
  return (
    <Container className={className}>
      <Header>{title}</Header>

      <ListContainer>
        {pokemonList.map(({ name, id }, index) => (
          <Pokemon key={`${id}-${index}`} onClick={() => onPokemonClick(id)}>
            {capitalize(name)}
          </Pokemon>
        ))}
      </ListContainer>
    </Container>
  );
}

function PokemonListContainer({ title, pokemonList, className, emptyMessage, onPokemonClick }) {
  return (
    <div className={className}>
      {isEmpty(pokemonList) ? (
        <div>{emptyMessage}</div>
      ) : (
        <PokemonList title={title} pokemonList={pokemonList} onPokemonClick={onPokemonClick} />
      )}
    </div>
  );
}

const Container = styled('div')({
  fontSize: '18px',
  backgroundColor: '#00d1b2',
  color: 'white',
  padding: '10px 20px',
  borderRadius: '10px',
});

const Pokemon = styled('div')({
  fontSize: '20px',
  marginLeft: '10px',
  marginRight: '10px',
  ':hover': {
    cursor: 'pointer',
  },
});

const Header = styled('h2')({
  fontSize: '18px',
});

const ListContainer = styled('div')({
  fontSize: '18px',
  display: 'flex',
  flexWrap: 'wrap',
});

export default PokemonListContainer;
