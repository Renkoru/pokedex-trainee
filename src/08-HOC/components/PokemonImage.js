// import React from 'react';
import styled from '@emotion/styled';

const widthMap = {
  default: '20%',
  medium: '150px',
};

const PokemonImage = styled.img`
  width: ${({ size }) => widthMap[size]};
`;

PokemonImage.defaultProps = {
  size: 'default',
};

export default PokemonImage;

// export default function PokemonImage({
//   src,
//   size = 'default',
//   ...rest
// }) {
//   const width = ;

//   return (
//     <img
//       src={src}
//       style={{
//         width,
//       }}
//       {...rest}
//     />
//   );
// }
