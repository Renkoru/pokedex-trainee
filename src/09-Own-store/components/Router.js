import React from 'react';

function Router({ children }) {
  const pathname = window.location.pathname;

  let result = null;

  React.Children.forEach(children, child => {
    if (child.props.path === pathname) {
      result = child;
    }
  });

  return <>{result}</>;
}

export default Router;
