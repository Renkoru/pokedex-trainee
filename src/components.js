import {
    Card,
    Box,
    Flex,
    Image,
    Heading,
} from 'rebass';

import Button from './components/Button.jsx';


export {
    Button,

    Card,
    Box,
    Flex,
    Image,
    Heading,
}
