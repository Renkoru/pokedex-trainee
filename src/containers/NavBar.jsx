import React from 'react';
import { connect } from 'react-redux';
import { Link } from "react-router-dom";

import { routes } from '../constants';
import { Button, Flex, Heading } from '../components';


export function NavBarComponent({ currentPlayer, errorMessage = '' }) {
    const navBarLinks = { marginRight: '15px' };

    return (
        <nav css={{ backgroundColor: '#92ef9d', padding: '2px 5px' }}>
          <Flex alignItems="center">
            <Heading>Players net. {errorMessage}</Heading>

            <ul css={{ listStyleType: 'none', display: 'flex' }}>
              <li css={navBarLinks}>
                <Link to={routes.home}>Home</Link>
              </li>
              <li css={navBarLinks}>
                <Link to={routes.players}>Players</Link>
              </li>
              <li css={navBarLinks}>
                <Link to={routes.admin}>Admin</Link>
              </li>
            </ul>

            <Heading ml="auto" mr="20px">
              {currentPlayer.name} : {currentPlayer.age} ({currentPlayer.location})
            </Heading>
          </Flex>
        </nav>
    );
}

function mapStateToProps(state) {
    const { playersState: { players, currentPlayerId } } = state;
    let currentPlayer = {};

    if (!!currentPlayerId) {
        currentPlayer =  players.find(({ stringId }) => stringId === currentPlayerId);
    }

    return {
        currentPlayer,
        errorMessage: state.monstersState.errorMessage,
    };
}


export default connect(mapStateToProps)(NavBarComponent);
