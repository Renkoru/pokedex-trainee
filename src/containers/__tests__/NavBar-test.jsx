import { NavBarComponent } from "../NavBar.jsx";
import { Heading } from '../../components';
import { shallow } from 'enzyme';


describe('NavBar', () => {
    it('renders app title correctly', () => {
        const props = {
            currentPlayer: {
                name: 'Lilly',
            }
        };

        const component = shallow(<NavBarComponent {...props} />);

        expect(component.find(Heading).at(0).text()).toEqual('Players net.');
    });

    it('renders correctly', () => {
        const props = {
            currentPlayer: {
                name: 'Lilly',
                location: 'Moscow'
            }
        };

        const component = shallow(<NavBarComponent {...props} />);

        expect(component).toMatchSnapshot();
    });
});
