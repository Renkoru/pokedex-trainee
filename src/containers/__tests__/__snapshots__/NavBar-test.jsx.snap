// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<nav
  css={
    Object {
      "backgroundColor": "#92ef9d",
      "padding": "2px 5px",
    }
  }
>
  <Styled(styled.div)
    alignItems="center"
  >
    <Styled(Styled(styled.div))
      as="h2"
      fontSize={4}
      fontWeight="bold"
      m={0}
    >
      Players net.
    </Styled(Styled(styled.div))>
    <ul
      css={
        Object {
          "display": "flex",
          "listStyleType": "none",
        }
      }
    >
      <li
        css={
          Object {
            "marginRight": "15px",
          }
        }
      >
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li
        css={
          Object {
            "marginRight": "15px",
          }
        }
      >
        <Link
          replace={false}
          to="/players"
        >
          Players
        </Link>
      </li>
      <li
        css={
          Object {
            "marginRight": "15px",
          }
        }
      >
        <Link
          replace={false}
          to="/admin"
        >
          Admin
        </Link>
      </li>
    </ul>
    <Styled(Styled(styled.div))
      as="h2"
      fontSize={4}
      fontWeight="bold"
      m={0}
      ml="auto"
      mr="20px"
    >
      Lilly
       : 
       (
      Moscow
      )
    </Styled(Styled(styled.div))>
  </Styled(styled.div)>
</nav>
`;
